dashboard/src/components/jobs/parameters/DataParamHandler.vue

```diff
@@ -0,0 +1,159 @@
+        {
+          calculate: "predicted performance" 
+        }
+
+  Emits a `new-data-params` event that contains an object that should be merged into
+  the root job parameters object.
+    Example event payloads:
+      {
+        weather_granularity: "system",
+        irradiance_type: "standard",
+        temperature_type: "cell"
+      }
+    Or:
+      {
+        predicted_data_parameters: {
+          data_available: "weather only",
+          weather_granularity: "system",
+          irradiance_type: "standard",
+          temperature_type: "cell"
+        },
+        actual_data_parameters: {
+          weather_granularity: "system",
+          irradiance_type: "standard",
+          temperature_type: "cell",
+          performance_granularity: "system"
+        }
+      }
+
+-->
+<template>
+  <div class="data-param-handler">
+    <div class="grid-container my-1">
+      <data-params
+        v-for="dataType of requiredDataParams"
+        :key="dataType"
+        :dataType="dataType"
+        :jobClass="jobClass"
+        @new-data-params="setDataParams"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch } from "vue-property-decorator";
+
+import DataParams from "@/components/jobs/parameters/DataParams.vue";
+
+Vue.component("data-params", DataParams);
+
+@Component
+export default class DataParamHandler extends Vue {
+  @Prop() jobTypeParams!: Record<string, string>;
+  parameters!: Record<string, any>;
+  data() {
+    return {
+      parameters: {}
+    };
+  }
+
+  mounted() {
+    this.emitParams();
+  }
+
+  setDataParams({
+    type,
+    parameters
+  }: {
+    type: string;
+    parameters: Record<string, string>;
+  }) {
+    if (type == "data_parameters") {
+      // Generic data_parameters type indicates that the data parameters should
+      // be added to the root level job specification.
+      this.parameters = parameters;
+    } else {
+      this.parameters[type] = parameters;
+    }
+    this.emitParams();
+  }
+
+  get isMonthly() {
+    return (
+      "compare" in this.jobTypeParams &&
+      this.jobTypeParams["compare"].includes("monthly")
+    );
+  }
+
+  get jobClass() {
+    if ("compare" in this.jobTypeParams) {
+      return "compare";
+    } else if (
+      this.jobTypeParams["calculate"] == "weather-adjusted performance ratio"
+    ) {
+      return "calculatepr";
+    } else {
+      return "calculate";
+    }
+  }
+  get requiredDataParams() {
+    if (this.isMonthly) {
+      return [];
+    }
+    if (this.jobClass == "calculate") {
+      if (this.jobTypeParams["calculate"] == "predicted performance") {
+        return ["predicted"];
+      } else {
+        return ["expected"];
+      }
+    } else if (this.jobClass == "compare") {
+      if (this.jobTypeParams["compare"] == "expected and actual performance") {
+        // Single set of "data parameters" for both expected and actual performance
+        return ["expected and actual"];
```

Comment:
should only be actual as all expected data is calculate from actual

Reply:
I've adjusted this to be a little more clear about this values function. I needed/wanted a special flag value to represent one set of data parameters for the whole job, as are used for the `expected and actual performance` compare job. I've changed this flag value to `all_data` and added a `<slot>` around the generated header in the DataParams component, so that the header can be overriden.